### ISSUE TITLE
Missing file during compilation

### ISSUE THREAD

**Issue by Meekohi** [openMVG/openMVG#226]
on `develop` branch dbcd1d76e3cd993c23a3087f94870f6d948a4865

```
/Users/meekohi/openMVG/src/openMVG/image/image_io.cpp:15:12: fatal error: 'tiffio.h' file not found
  #include "tiffio.h"
           ^
1 error generated.

** BUILD FAILED **
```

I'm on OSX. `libtiff-4.0.3 already installed`.

**Comment by pmoulon** [comment]
can you locate if you have a tiffio.h file on your machine?

**Comment by Meekohi** [comment]
```
meekohi ~/openMVG_Build$ ls /usr/local/Cellar/libtiff/4.0.3/include/
tiff.h     tiffconf.h tiffio.h   tiffio.hxx tiffvers.h
```

**Comment by Meekohi** [comment]
Ah it's alos in /thirdparty. Probably just need to tweak the CMake file (wish I understood how CMake worked better...)

```
meekohi ~/openMVG_Build$ ls ~/openMVG/src/third_party/tiff/
CMakeLists.txt       tif_close.c          tif_dirinfo.c        tif_flush.c          tif_ojpeg.c          tif_strip.c          tif_zip.c
COPYRIGHT            tif_codec.c          tif_dirread.c        tif_getimage.c       tif_open.c           tif_swab.c           tiff.h
ChangeLog            tif_color.c          tif_dirwrite.c       tif_jbig.c           tif_packbits.c       tif_thunder.c        tiffconf.vc.h
RELEASE-DATE         tif_compress.c       tif_dumpmode.c       tif_jpeg.c           tif_pixarlog.c       tif_tile.c           tiffconf.wince.h
VERSION              tif_config.h.cmakein tif_error.c          tif_jpeg_12.c        tif_predict.c        tif_unix.c           tiffio.h
libtiff.def          tif_config.vc.h      tif_extension.c      tif_luv.c            tif_predict.h        tif_version.c        tiffio.hxx
mkg3states.c         tif_config.wince.h   tif_fax3.c           tif_lzma.c           tif_print.c          tif_warning.c        tiffiop.h
t4.h                 tif_dir.c            tif_fax3.h           tif_lzw.c            tif_read.c           tif_win32.c          tiffvers.h
tif_aux.c            tif_dir.h            tif_fax3sm.c         tif_next.c           tif_stream.cxx       tif_write.c          uvcode.h
```

**Comment by pmoulon** [comment]
On mac TIFF lib of the computer must not be used:
See here, and by default the tiff openMVG directory must be used https://github.com/openMVG/openMVG/blob/develop/src/CMakeLists.txt#L87
So perhaps a strange include behavior is happening on your computer.

**Comment by pmoulon** [comment]
Normally tiff include library is handled here
https://github.com/openMVG/openMVG/blob/develop/src/openMVG/image/CMakeLists.txt#L20

Perhaps ${TIFF_LIBRARIES} is empty for you...

**Comment by rperrot** [comment]
I've got the same issue. (on OS X too) 

TIFF_INCLUDE_DIR is not empty but doesn't get added to OpenMVG_INCLUDES but I can't understand why ... . So when building, it doesn't search the correct include. 

printing TIFF_INCLUDE_DIR gives correct results. Maybe it's a cmake bug ?

**Comment by rperrot** [comment]
Ok, I've got the solution !

@Meekohi You probably built openMVG first using master branch then built it with develop branch (that's what I did). Since you probably did it in the same directory (that's what I did too !), CMakeCache.txt is present and contains the old (cached) OpenMVG_INCLUDES variable. Simply remove CMakeCache.txt then relaunch cmake generation and you'll have correct compilation. 

@pmoulon The issue is related to the CACHE PATH option of OpenMVG_INCLUDES in main CMakeLists.txt, when disabled the variable gets rewrited each time you run cmake and also solve the problem. I don't know if it's better removing from the CMakeLists.txt file or adding a comment to the user that would build from master then from develop branch.
